transforms/universal/pdf2md/python/src/pdf2md_transform.py

```diff
@@ -0,0 +1,346 @@
+#         :param files - list of files to process
+#         :return: dictionary of filter init params
+#         """
+
+#         # This cannot yet be done, because the converter is not easily serializable
+
+#         # self.models_dir = Path(self.params.get(ingest_modelsdir_key, "scratch"))
+#         # print("Downloading models")
+#         # logger.info("Downloading models")
+#         # DocumentConverter.download_models(
+#         #     artifacts_path=self.models_dir,
+#         #     artifacts_url=DocumentConverter.artifacts_urls["standard"],
+#         # )
+#         # print("Initializing converter")
+#         # logger.info("Initializing converter")
+#         # converter = DocumentConverter(artifacts_path=self.models_dir)
+#         # converter_refs = ray.put(converter)
+
+#         # logger.debug(f"Placed converter into Ray object storage under reference {converter_refs=}")
+#         # return {ingest_converter_refname: converter_refs} | self.params
+
+#         return self.params
+
+
+class PdfToMdTransformConfiguration(TransformConfiguration):
+    """
+    Provides support for configuring and using the associated Transform class include
+    configuration with CLI args and combining of metadata.
+    """
+
+    def __init__(self):
+        super().__init__(
+            name=shortname,
+            transform_class=PdfToMdTransform,
+            # remove_from_metadata=[ingest_data_factory_key],
+        )
+        self.daf = None
+
+    def add_input_params(self, parser: ArgumentParser) -> None:
+        """
+        Add Transform-specific arguments to the given parser.
+        By convention a common prefix should be used for all mutator-specific CLI args
+        (e.g, noop_, pii_, etc.)
+        """
+        parser.add_argument(
+            f"--{pdf2md_modelsdir_key}",
+            type=str,
+            help="Path where to download models",
+            default="scratch",
+        )
+        parser.add_argument(
+            f"--{pdf2md_download_models_key}",
+            type=bool,
+            help="If true, models will be downloaded locally. Not needed when inside the docker image.",
+            default=False,
+        )
+        parser.add_argument(
+            f"--{pdf2md_do_table_structure_key}",
+            type=str2bool,
+            help="If true, detected tables will be processed with the table structure model.",
+            default=True,
+        )
+        parser.add_argument(
+            f"--{pdf2md_do_ocr_key}",
+            type=str2bool,
+            help="If true, optical character recognization (OCR) will be used to read the PDF content.",
+            default=False,
+        )
+
+    def apply_input_params(self, args: Namespace) -> bool:
+        """
+        Validate and apply the arguments that have been parsed
+        :param args: user defined arguments.
+        :return: True, if validate pass or False otherwise
+        """
+        dargs = vars(args)
+        self.params = {
```

Comment:
For the de-prefixing, see what noop does https://github.com/IBM/data-prep-kit/blob/736055c99c85e77e76cf5939c7cd124320a6c44e/transforms/universal/noop/python/src/noop_transform.py#L114

Reply:
thanks for the pointers, we don't know about it. should be there

Reply:
done